pkg/plugin/driver.go

```diff
@@ -158,9 +162,17 @@ func (h *Clickhouse) Macros() sqlds.Macros {
 	}
 }
 
-func (h *Clickhouse) Settings(backend.DataSourceInstanceSettings) sqlds.DriverSettings {
+func (h *Clickhouse) Settings(config backend.DataSourceInstanceSettings) sqlds.DriverSettings {
+	settings, err := LoadSettings(config)
+	timeout := 30
+	if err == nil {
+		t, err := strconv.Atoi(settings.Timeout)
+		if err == nil {
+			timeout = t
+		}
+	}
 	return sqlds.DriverSettings{
```

Comment:
@bossinc added this. We also need to impose a LIMIT on queries I think or at least the number of rows read